src/app/components/jobs-wrapper/job-logs/job-logs.component.ts

```diff
@@ -0,0 +1,59 @@
+import {Component, OnInit, OnDestroy, ElementRef, ViewChild, AfterViewInit, Input } from '@angular/core';
+import { WebSocketLogsService } from '@services/web-socket-logs.service';
+
+@Component({
+  selector: 'mist-job-logs',
+  templateUrl: './job-logs.component.html',
+  styleUrls: ['./job-logs.component.scss'],
+  providers: [WebSocketLogsService],
+  host: {
+    '(document:keydown)': 'handleKeyboardEvents($event)'
+  }
+})
+export class JobLogsComponent implements OnInit, AfterViewInit, OnDestroy {
+  @ViewChild('jobLogs') elem: ElementRef;
+  @Input() jobId: number;
+  parent: Element;
+  isFullScreenEnabled: Boolean;
+  logs: Array<object> = [];
+  subscriber: any;
+
+  constructor(WebSocketLogsService: WebSocketLogsService) {
+    this.subscriber = WebSocketLogsService;
+  }
+
+  ngOnInit() {
+    this.subscriber.connect(this.jobId).subscribe((data) => {
+      this.logs.push(data);
+    });
+  }
+
+  ngAfterViewInit() {
+    this.parent = this.elem.nativeElement.parentElement;
+  }
+
+  toggleToFullScreen() {
+    if (!this.isFullScreenEnabled) {
+      this.elem.nativeElement.classList.toggle('job-logs--full-screen');
+      document.body.appendChild(this.elem.nativeElement);
+      this.isFullScreenEnabled = true;
+      location.hash = '';
+    } else {
+      this.parent.appendChild(this.elem.nativeElement);
+      this.elem.nativeElement.classList.toggle('job-logs--full-screen');
+      this.isFullScreenEnabled = false;
+      location.hash = '#logs';
+    }
+  }
+
+  handleKeyboardEvents(event: KeyboardEvent) {
+    if (event.keyCode === 27) {
+      this.toggleToFullScreen();
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.subscriber.disconnect().unsubscribe();
```

Comment:
i think it will be better to separate `disconnect()` and `subscribe()`.
for example
```
this.WebSocketLogsService.disconnect();
this.subscriber.unsubscribe();
```
What do you think about this?